stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt

```diff
@@ -278,18 +301,28 @@ internal class PaymentSheetActivity : BaseSheetActivity<PaymentResult>() {
         viewModel.updateMode(transitionTarget.sheetMode)
     }
 
-    private fun setupBuyButton() {
-        viewBinding.buyButton.completedAnimation.observe(this) { completedState ->
-            completedState?.paymentIntentResult?.let { paymentIntentResult ->
-                onActionCompleted(paymentIntentResult)
+    private val viewStateObserver = { viewState: ViewState.PaymentSheet? ->
+        viewBinding.buyButton.updateState(
+            state = when (viewState) {
+                null -> null
+                is ViewState.PaymentSheet.Ready -> PrimaryButton.State.Ready(
+                    getLabelText(viewState)
+                )
+                is ViewState.PaymentSheet.StartProcessing -> PrimaryButton.State.StartProcessing
+                is ViewState.PaymentSheet.FinishProcessing -> PrimaryButton.State.FinishProcessing
+            },
+            completeCallback = {
+                onActionCompleted(
+                    (viewState as ViewState.PaymentSheet.FinishProcessing).result
+                )
             }
```

Comment:
This will crash when `viewState != ViewState.PaymentSheet.FinishProcessing`

Reply:
I think this is OBE